### PR TITLE
Expiry date , trim space and phone verification

### PR DESCRIPTION
**Pull request by aishwa8141**

### CODE REVIEW

File: csvProcessor/src/main/java/org/incredible/csvProcessor/CertModelFactory.java [hunk 1]
```diff
@@ -20,14 +25,19 @@ public CertModelFactory(HashMap<String, String> csvprop) {
...
+        String expiry;
         try {
             CertModel certModel = new CertModel();
             for (HashMap.Entry<String, String> member : csvProperties.entrySet()) {
                 String methodName = "set" + member.getKey();
                 Class params = getParamType(methodName);
                 Method method = certModel.getClass().getMethod(methodName, params);
                 method.setAccessible(true);
-                method.invoke(certModel, csvRecord.get(member.getValue()));
+                if (member.getValue().equals("Expiry")) {
```
Comment (indrajra) [inline csvProcessor/src/main/java/org/incredible/csvProcessor/CertModelFactory.java]: why look at Value...? Can't you look at Key and make this if clause.

File: csvProcessor/src/main/java/org/incredible/csvProcessor/CertModelFactory.java [hunk 2]
```diff
@@ -53,6 +63,43 @@ private Class getParamType(String methodName) {
         } else
             return parameterTypes[0];
     }
+
+
+    private String calculateExpiryDateOfCertificate(String expiryDate, String issueDate) {
```
Comment (indrajra) [inline csvProcessor/src/main/java/org/incredible/csvProcessor/CertModelFactory.java]: Create a class called "ExpiryDateValuator" that has constructor taking up issueDate. Then move this logic to a function, 
`String getExpiryDate(String expiryDateValue)`

Also, as the sheet indicates, it must work for lowercase and uppercase 2m and 2M.

File: csvProcessor/src/main/java/org/incredible/csvProcessor/Main.java [hunk 3]
```diff
@@ -215,6 +215,13 @@ private static void generateHtmlTemplateForCertificate(Assertion assertion) {
 
     }
 
+    private static void sendSmsToUSer(Assertion assertion) {
+        SMSDeliver smSdeliver = new SMSDeliver();
```
Comment (indrajra) [inline csvProcessor/src/main/java/org/incredible/csvProcessor/Main.java]: What's this? We haven't implemented and there are no classes in this PR.

File: certgen/certProcessor/src/main/java/org/incredible/certProcessor/CertificateFactory.java [hunk 4]
```diff
@@ -43,8 +42,9 @@ public CertificateExtension createCertificate(CertModel certModel, String contex
         /**
          *  recipent object
          *  **/
-        compositeIdentityObjectBuilder.setName(certModel.getRecipientName()).setId(uuid).setHashed(true).
-                setType(new String[]{"urn"});
+        compositeIdentityObjectBuilder.setName(certModel.getRecipientName()).setId(certModel.getRecipientPhone())
+                .setHashed(true).
```
Comment (indrajra) [inline certgen/certProcessor/src/main/java/org/incredible/certProcessor/CertificateFactory.java]: hash must be true only when phone number is hashed (not visible as plain phone number).

**Review by indrajra — CHANGES_REQUESTED** [review]

File: csvProcessor/src/main/java/org/incredible/csvProcessor/Main.java (hunk 3 above)
Comment (aishwa8141) [reply csvProcessor/src/main/java/org/incredible/csvProcessor/Main.java]: I was trying to send sms, I did commit that class

File: certgen/certProcessor/src/main/java/org/incredible/builders/AssessmentBuilder.java [hunk 5]
```diff
@@ -4,7 +4,7 @@
 
 public class AssessmentBuilder implements IBuilder<Assessment> {
 
-    Assessment assessment;
+    Assessment assessment = new Assessment();
```
Comment (indrajra) [inline certgen/certProcessor/src/main/java/org/incredible/builders/AssessmentBuilder.java:7]: Put a TODO.

File: certgen/certProcessor/src/main/java/org/incredible/builders/CertificateExtensionBuilder.java [hunk 6]
```diff
@@ -77,8 +78,20 @@ public CertificateExtensionBuilder setExpires(String expires) {
...
+    }
+
+
     @Override
     public CertificateExtension build() {
-        return this.certificateExtension;
+        return certificateExtension;
     }
+
+//    @Override
```
Comment (indrajra) [inline certgen/certProcessor/src/main/java/org/incredible/builders/CertificateExtensionBuilder.java:92]: remove

File: certgen/certProcessor/src/main/java/org/incredible/certProcessor/CertificateFactory.java [hunk 7]
```diff
@@ -27,46 +28,63 @@
...
-        uuid = UUID.randomUUID().toString();
-        System.out.println(uuid);
+
+        uuid = "http://localhost:8080/_schemas/Certificate.json";
+
         CertificateExtensionBuilder certificateExtensionBuilder = new CertificateExtensionBuilder(context);
         CompositeIdentityObjectBuilder compositeIdentityObjectBuilder = new CompositeIdentityObjectBuilder(context);
         BadgeClassBuilder badgeClassBuilder = new BadgeClassBuilder(context);
-        AssessedEvidenceBuilder assessedEvidenceBuilder = new AssessedEvidenceBuilder(context);
+        AssessedEvidenceBuilder assessedEvidenceBuilder = new AssessedEvidenceBuilder("http://localhost:8080/_schemas/Extensions/AssessedEvidence/Context.json");
```
Comment (indrajra) [inline certgen/certProcessor/src/main/java/org/incredible/certProcessor/CertificateFactory.java:37]: Give only param - context.

**Review by indrajra — CHANGES_REQUESTED** [review]

**Pull request merged**